sdk-wallet/keystores/encrypted_keystore.md

```diff
@@ -1,120 +1,52 @@
 ## EncryptedKeystore
 
+`EncryptedKeystore` handles encrypted JSON files.
+
+Note: current design is suboptimal. `EncryptedKeystore` handles both legacy keystore files (that hold encrypted secret keys) and new keystore files (that hold encrypted mnemonics). `get_kind()` can be used as a differentiator. A future design might involve two separate classes.
+
 ```
 class EncryptedKeystore:
     // The constructor is not captured by the specs; it's up to the implementing library to define it.
+    // Instance fields to be held (suggestion): 
+    //  - keys_computer: IKeysComputer
+    //  - kind: "secretKey|mnemonic"
+    //  - secretKey: ISecretKey (will be nil if kind == 'mnemonic')
+    //  - mnemonic: Mnemonic (will be nil if kind == 'secretKey')
 
     // Named constructor
-    static new_from_secret_key(secret_key: ISecretKey): EncryptedKeystore
+    // This should have a trivial implementation.
+    static new_from_secret_key(keys_computer: IKeysComputer, secret_key: ISecretKey): EncryptedKeystore
 
     // Named constructor
-    // Below, "wallet_provider" should implement "derive_secret_key_from_mnemonic()".
-    // Advice: in the implementation all the parameters will be held as instance state (private fields).
-    static new_from_mnemonic(wallet_provider: IWalletProvider, mnemonic: Mnemonic): EncryptedKeystore
+    // This should have a trivial implementation.
+    static new_from_mnemonic(keys_computer: IKeysComputer, mnemonic: Mnemonic): EncryptedKeystore
 
     // Importing "constructor"
-    static import_from_object(wallet_provider: IWalletProvider, object: KeyfileObject, password: string): EncryptedKeystore
+    // This should decrypt the encrypted data, then call the (private) constructor.
+    // "password" should not be retained.
+    static import_from_object(keys_computer: IKeysComputer, object: KeyfileObject, password: string): EncryptedKeystore
```

Comment:
`object: KeyfileObject` should be something like `keystore_content: EncryptedKeystoreContent`